client/src/components/Layout/Layout.js

```diff
@@ -0,0 +1,203 @@
+                    <main>
+                        <Navbar to={`/member-page/${props.userid}`} profileImage={props.image} page={2}></Navbar>
+                        <ForumPageHashtag
+                            handle_logout={props.handleLogout}
+                            {...routerProps}
+                            {...props}
+                        />
+                    </main>
+                )}
+            ></Route>
+            <Route
+                path="/forum-page-discover"
+                exact
+                render={routerProps => (
+                    <main>
+                        <Navbar to={`/member-page/${props.userid}`} profileImage={props.image} page={2}></Navbar>
+                        <ForumPageDiscover
+                            handle_logout={props.handleLogout}
+                            {...routerProps}
+                            {...props}
+                        />
+                    </main>
+                )}
+            ></Route>
+            <Route
+                path="/forum-page-topic"
+                exact
+                render={routerProps => (
+                    <main>
+                        <Navbar to={`/member-page/${props.userid}`} profileImage={props.image} page={2}></Navbar>
+                        <ForumPageTopic handle_logout={props.handleLogout} {...routerProps} {...props} />
+                    </main>
+                )}
+            ></Route>
+            <Route
+                path="/directory/people"
+                exact
+                render={routerProps => (
+                    <main>
+                        <Navbar to={`/member-page/${props.userid}`} profileImage={props.image} page={0}></Navbar>
+                        <DirectoryPeople handle_logout={props.handleLogout} {...routerProps} />
+                    </main>
+                )}
+            ></Route>
+            <Route
+                path="/directory/groups"
+                exact
+                render={routerProps => (
+                    <main>
+                        <Navbar to={`/member-page/${props.userid}`} profileImage={props.image} page={0}></Navbar>
+                        <DirectoryGroup handle_logout={props.handleLogout} {...routerProps} />
+                    </main>
+                )}
+            ></Route>
+            <Route
+                path="/login"
+                exact
+                render={routerProps => (
+                    <LoginPage handleLogin={props.handleLogin} {...routerProps} {...props} />
+                )}
+            ></Route>
+            <Route
+                path="/register"
+                exact
+                render={routerProps => (
+                    <OnboardingPage
+                        handleSignup={props.handleSignup}
+                        {...routerProps}
+                        logged_in={props.logged_in}
+                    />
+                )}
+            ></Route>
```

Comment:
All the routing should stay in `App`, but be nested inside the `Layout` component. See #665 for what I mean

Reply:
Refactored using props.children.